modules/renter/contractor/contractor.go

```diff
@@ -0,0 +1,154 @@
+package contractor
+
+import (
+	"errors"
+	"os"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+var (
+	errNilCS           = errors.New("cannot create contractor with nil consensus set")
+	errNilWallet       = errors.New("cannot create contractor with nil wallet")
+	errNilTpool        = errors.New("cannot create contractor with nil transaction pool")
+	errUnknownContract = errors.New("no record of that contract")
+)
+
+// a Contract includes the original contract made with a host, along with
+// the most recent revision.
+type Contract struct {
+	IP              modules.NetAddress
```

Comment:
You should identify hosts by their public keys, not by their ip addresses, because the ip addresses can change and we plan to add support for dynmaic ips, hosts will update renters through either the blockchain or through some type of peer protocol.